src/overlords/offense/controllerAttacker.ts

```diff
@@ -0,0 +1,106 @@
+// Controller attacker overlord.  Spawn CLAIM creeps to mass up on a controller and attack all at once
+
+import {Zerg} from '../../zerg/Zerg';
+import {OverlordPriority} from '../../priorities/priorities_overlords';
+import {Overlord} from '../Overlord';
+import {DirectiveControllerAttack} from '../../directives/offense/controllerAttack';
+import {CreepSetup} from '../CreepSetup';
+import {profile} from '../../profiler/decorator';
+import {log} from '../../console/log';
+import {SpawnGroup} from 'logistics/SpawnGroup';
+
+const ControllerAttacker = new CreepSetup('ctrlAttack', {
+	pattern: [CLAIM, MOVE],
+	sizeLimit: Infinity,
+	ordered: true
+});
+
+interface ControllerAttackerMemory extends CreepMemory {
+	attackPos: protoPos;
+}
+
+@profile
+export class ControllerAttackerOverlord extends Overlord {
+
+	attackers: ControllerAttackerZerg[];
+	scouts: Zerg[];
+	attackPos: RoomPosition[] | undefined;
+	spawns: StructureSpawn[] | undefined;
+	readyAtTick: number;
+
```

Comment:
`spawnGroup: SpawnGroup`, since you definitely have roomName it can be definitely typed